packages/trader/src/AppV2/Components/Filter/time-filter.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import Chip from 'AppV2/Components/Chip';
+import { toMoment } from '@deriv/shared';
+import { ActionSheet, RadioGroup } from '@deriv-com/quill-ui';
+import { Localize } from '@deriv/translations';
+import CustomDateFilterButton from './custom-time-filter-button';
+import DateRangePicker from 'AppV2/Components/DatePicker';
+
+type TTimeFilter = {
+    customTimeRangeFilter?: string;
+    handleDateChange: (values: { to?: moment.Moment; from?: moment.Moment; is_batch?: boolean }) => void;
+    setTimeFilter: (newTimeFilter?: string | undefined) => void;
+    setCustomTimeRangeFilter: (newCustomTimeFilter?: string | undefined) => void;
+    setNoMatchesFound: React.Dispatch<React.SetStateAction<boolean>>;
+    timeFilter?: string;
+};
+
+const timeFilterList = [
+    {
+        value: '0',
+        label: <Localize i18n_default_text='All time' />,
+    },
+    {
+        value: 'Today',
+        label: <Localize i18n_default_text='Today' />,
+    },
+    {
+        value: 'Yesterday',
+        label: <Localize i18n_default_text='Yesterday' />,
+    },
+    {
+        value: '7',
+        label: <Localize i18n_default_text='Last 7 days' />,
+    },
+    {
+        value: '30',
+        label: <Localize i18n_default_text='Last 30 days' />,
+    },
+    {
+        value: '60',
+        label: <Localize i18n_default_text='Last 60 days' />,
+    },
+    {
+        value: '90',
+        label: <Localize i18n_default_text='Last 90 days' />,
+    },
+];
+
+const TimeFilter = ({
+    customTimeRangeFilter,
+    handleDateChange,
+    setTimeFilter,
+    setCustomTimeRangeFilter,
+    setNoMatchesFound,
+    timeFilter,
+}: TTimeFilter) => {
+    const [isDropdownOpen, setIsDropdownOpen] = React.useState(false);
+    const [showDatePicker, setShowDatePicker] = React.useState(false);
+
+    const defaultCheckedTime = '0';
+    const selectedRadioButtonValue = customTimeRangeFilter || timeFilter || defaultCheckedTime;
+    const isChipSelected = !!(customTimeRangeFilter || timeFilter);
+
+    const onRadioButtonChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const value = e.target.value;
+        if (value === defaultCheckedTime) {
+            onReset();
+            return;
+        }
+        setTimeFilter(value);
+        setIsDropdownOpen(false);
+
+        if (value === 'Today') {
```

Comment:
may not work when translated

Reply:
value is taken from const `timeFilterList`, where it's just string without <Localize/>
<img width="749" alt="Screenshot 2024-05-29 at 9 00 11 AM" src="https://github.com/binary-com/deriv-app/assets/121025168/8dfcb2e3-9e85-4b25-a269-abf2fdad1d3b">